src/vs/workbench/contrib/preferences/browser/settingsTree.ts

```diff
@@ -75,6 +76,88 @@ function getExcludeDisplayValue(element: SettingsTreeSettingElement): IListDataI
 		});
 }
 
+function getEnumOptionsFromSchema(schema: IJSONSchema): IObjectEnumOption[] {
+	const enumDescriptions = schema.enumDescriptions ?? [];
+
+	return (schema.enum ?? []).map((value, idx) => {
+		const description = idx < enumDescriptions.length
+			? enumDescriptions[idx]
+			: undefined;
+
+		return { value, description };
+	});
+}
+
+function getObjectDisplayValue(element: SettingsTreeSettingElement): IObjectDataItem[] {
+	const data = element.isConfigured ?
+		{ ...element.defaultValue, ...element.scopeValue } :
+		element.defaultValue;
+
+	let items: IObjectDataItem[] = [];
+
+	if (element.setting.objectProperties) {
+		const wellDefinedKeys = Object.keys(element.setting.objectProperties);
+		const keyOptions = wellDefinedKeys.map(value => ({ value }));
+
+		items = items.concat(
+			wellDefinedKeys
+				.filter(key => !!data[key])
+				.map(key => {
+					const valueEnumOptions = getEnumOptionsFromSchema(element.setting.objectProperties![key]);
+
+					return {
+						key: {
+							type: 'enum',
+							data: key,
+							options: keyOptions,
+						},
+						value: {
+							type: valueEnumOptions.length > 0 ? 'enum' : 'string',
+							data: data[key],
+							options: valueEnumOptions,
+						},
+					};
+				})
+		);
+	}
+
+	if (element.setting.objectPatternProperties) {
+		const patternsAndSchemas = Object
+			.entries(element.setting.objectPatternProperties)
+			.map(([pattern, schema]) => ({
+				pattern: new RegExp(pattern),
+				schema
+			}));
+
+		const keysWithSchema = Object.keys(data)
+			.filter(key => !!data[key] && !(key in (element.setting.objectProperties ?? {})))
```

Comment:
Why filter on value? This will also exclude empty strings

Reply:
you're right, we should only remove values that are null or undefined.

Reply:
That shouldn't typically happen since you've filtered for settings where the value is only a string already right?